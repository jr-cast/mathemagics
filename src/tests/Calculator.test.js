import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Testing Calculator compoenent', () => {
  test('it renders all Calculator components correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Adding to numbers using fireEvent on calculator', () => {
    render(<Calculator />);
    fireEvent.click(screen.queryByText('7'));
    fireEvent.click(screen.queryByText('+'));
    fireEvent.click(screen.queryByText('7'));
    fireEvent.click(screen.queryByText('='));
    expect(screen.queryByText('14')).not.toBeNull();
  });

  test('Dividing numbers using fireEvent on calculator', () => {
    render(<Calculator />);
    fireEvent.click(screen.queryByText('7'));
    fireEvent.click(screen.queryByText('รท'));
    fireEvent.click(screen.queryByText('2'));
    fireEvent.click(screen.queryByText('='));
    expect(screen.queryByText('3.5')).not.toBeNull();
  });
});
