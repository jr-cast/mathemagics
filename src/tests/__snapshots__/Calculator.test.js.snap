// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator compoenent it renders all Calculator components correctly 1`] = `
<div
  className="main"
>
  <div
    className="grid-container"
  >
    <div
      className="grid-itemResult"
    >
      <span />
    </div>
    <button
      className="grid-item"
      onClick={[Function]}
      text="AC"
      type="button"
    >
      AC
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="+/-"
      type="button"
    >
      +/-
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="%"
      type="button"
    >
      %
    </button>
    <button
      className="grid-item operator"
      onClick={[Function]}
      text="รท"
      type="button"
    >
      รท
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="7"
      type="button"
    >
      7
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="8"
      type="button"
    >
      8
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="9"
      type="button"
    >
      9
    </button>
    <button
      className="grid-item operator"
      onClick={[Function]}
      text="x"
      type="button"
    >
      x
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="4"
      type="button"
    >
      4
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="5"
      type="button"
    >
      5
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="6"
      type="button"
    >
      6
    </button>
    <button
      className="grid-item operator"
      onClick={[Function]}
      text="-"
      type="button"
    >
      -
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="1"
      type="button"
    >
      1
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="2"
      type="button"
    >
      2
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="3"
      type="button"
    >
      3
    </button>
    <button
      className="grid-item operator"
      onClick={[Function]}
      text="+"
      type="button"
    >
      +
    </button>
    <button
      className="grid-zero"
      onClick={[Function]}
      text="0"
      type="button"
    >
      0
    </button>
    <button
      className="grid-item"
      onClick={[Function]}
      text="."
      type="button"
    >
      .
    </button>
    <button
      className="grid-item operator"
      onClick={[Function]}
      text="="
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
